compiler/rustc_builtin_macros/src/format_foreign.rs

```diff
@@ -513,7 +521,7 @@ pub(crate) mod printf {
                 '0'..='9' => {
                     let end = at_next_cp_while(next, char::is_ascii_digit);
                     state = Length;
-                    precision = Some(Num::from_str(at.slice_between(end).unwrap(), None));
+                    precision = Some(Num::from_str(at.slice_between(end).unwrap(), None).unwrap());
```

Comment:
```suggestion
                    precision = at.slice_between(end).map(|num| Num::from_str(num, None));
```

Reply:
should add `?` at end to convert `Option<Option<Num>>` to `Option<Num>`